src/main/java/org/terasology/climateConditions/HypothermiaSystem.java

```diff
@@ -62,11 +51,57 @@ public void modifySpeed(GetMaxSpeedEvent event, EntityRef player, HypothermiaCom
     }
 
     /**
-     * Reduces the jump speed of the player.
-     * Is only active iff the player has a {@link HypothermiaComponent}.
+     * Reduces the jump speed of the player. Is only active iff the player has a {@link HypothermiaComponent}.
      */
     @ReceiveEvent
     public void modifyJumpSpeed(AffectJumpForceEvent event, EntityRef player, HypothermiaComponent hypothermia) {
         event.multiply(hypothermia.jumpSpeedMultiplier);
     }
+
+    @ReceiveEvent
+    public void hypothermiaLevelChanged(HypothermiaLevelChangedEvent event, EntityRef player,
+                                        HypothermiaComponent hypothermia) {
+        modifySpeedMultipliers(player, hypothermia, event.getNewLevel());
+        //Adding New Effects when Hypothermia Level Increased.
+        switch (event.getNewLevel()) {
+            case 1:
+                if (event.getOldLevel() < 1) {
+                    visibleBreathingSystem.applyVisibleBreath(player);
+                }
+                break;
+            case 3:
+                if (event.getOldLevel() < 3) {
+                    frostbiteSystem.applyFrostbite(player);
+                }
+        }
+        //Removing effects when Hypothermia Level is decreased.
+        switch (event.getOldLevel()) {
+            case 1:
+                if (event.getNewLevel() < 1) {
+                    visibleBreathingSystem.removeVisibleBreath(player);
+                }
+                break;
+            case 3:
+                if (event.getNewLevel() < 3) {
+                    frostbiteSystem.removeFrostbite(player);
+                }
+        }
```

Comment:
If you do this based on increase or decrease, how about first checking which it is:
```
oldLvl = event.getOldLevel();
newLvl = event.getNewLevel();
if (newLvl > oldLvl) {
// increase
} else if (newLvl < oldLvl){
// decrease
}
```
and then switching only on the newLvl.
In the level increase case, we only need to add new effects and in the level decrease case, we can remove the effects of `newLvl+1` or alternatively simply remove all effects (removing non-existing effects should be a no-op) and re-applying the effects for newLvl.

Reply:
I specifically didn't want to remove effects from newLvl+1 just in case if some another module was to directly change the temperature level from criticalHigh to normal or low, etc.... the system shouldn't simply break... So I kept it the way it is.

And the newLevel, oldLevel check condition really made the code look a lot better. :thumbsup: